src/main/java/bookbob/functions/CommandHandler.java

```diff
@@ -792,6 +799,101 @@ public void removePastAppointments(AppointmentRecord appointmentRecord) throws I
         FileHandler.autosave(appointmentRecord);
     }
 
+    //@@author kaboomzxc
+    public void addVisit(String input, Records records) throws IOException {
+        try {
+            // Extract NRIC first (mandatory field)
+            int nricStart = input.indexOf("ic/");
+            if (nricStart == -1) {
+                System.out.println("Please provide the patient's NRIC.");
+                return;
+            }
+            int nricEnd = findNextFieldStart(input, nricStart + 3);
+            String nric = input.substring(nricStart + 3, nricEnd).trim();
+
+            // Find the patient with matching NRIC
+            Patient targetPatient = null;
+            for (Patient patient : records.getPatients()) {
+                if (patient.getNric().equals(nric)) {
+                    targetPatient = patient;
+                    break;
+                }
+            }
+
+            if (targetPatient == null) {
+                System.out.println("No patient found with NRIC: " + nric);
+                return;
+            }
+
+            // Extract visit date (mandatory field)
+            int visitStart = input.indexOf("v/");
+            if (visitStart == -1) {
+                System.out.println("Please provide the visit date and time.");
+                return;
+            }
+            int visitEnd = findNextFieldStart(input, visitStart + 2);
+            String visitDateString = input.substring(visitStart + 2, visitEnd).trim();
+
+            // Parse visit date
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+            LocalDateTime visitDate;
+            try {
+                visitDate = LocalDateTime.parse(visitDateString, formatter);
+            } catch (DateTimeParseException e) {
+                System.out.println("Invalid date format. Please use dd-MM-yyyy HH:mm format (e.g., 21-10-2024 15:48)");
+                return;
+            }
```

Comment:
Good catch of DateTimeParseException in case user inputs the wrong format for the visit date and time. I like how you include an example of date and time in the correct format too.